### DISCUSSION TITLE
Gradient is not being applied to text

### DISCUSSION THREAD

**Discussion by kaminski-dev** [tailwindlabs/tailwindcss#11885] [Help]
Hello, 

I have a vite project with tailwindcss and what I would like to do is, to add a gradient to a text which I have.
Bascially in the tailwindcss config I added this part:
/** @type {import('tailwindcss').Config} */
export default {
  content: ["./index.html", "./src/**/*.{js,ts,jsx,tsx}"],
  theme: {
    extend: {
      gradientColorStops: {
        "golden-gradient": [
          "#C29439",
          "#D9B879",
          "#765713",
          "#FFE69F",
          "#906420",
          "#D9BB7B",
          "#906210",
        ],
      },
I want my text to use the golden-gradient so below you can see how I tried to applying it:
            <li className="bg-gradient-to-r from-golden-gradient text-transparent bg-clip-text">
              Hello World
            </li>
But unfortunately the text doesn't change at all, it took just the color of the background which is black. 

Am I doing something wrong here or how should I approach this one? As well please let me know If I need to add more information.

my index.css

:root {
  font-synthesis: none;
  text-rendering: optimizeLegibility;
  background-color: #010203;
  overflow-x: hidden;
}

@tailwind base;
@tailwind components;
@tailwind utilities;

  "autoprefixer": "^10.4.15",
    "postcss": "^8.4.28",
    "tailwindcss": "^3.3.3",

**Comment by wongjn** [comment]
Can't quite see how you've tried to apply it. Consider providing a [Tailwind Play](https://play.tailwindcss.com) that reproduces the unexpected behavior.

**Reply by kaminski-dev** [reply]
Thank you very much for the fast response, here is the Tailwind play link:
https://play.tailwindcss.com/xoDD7HQCGt

**Reply by kaminski-dev** [reply]
What I am trying to achieve is to apply a linear gradient to an SVG. Here is the SVG code with the linear gradient:
<linearGradient id="id0" gradientUnits="userSpaceOnUse" x1="2651" y1="918" x2="3121" y2="509">
   <stop offset="0" style="stop-color:#C29439"/>
   <stop offset="0.14902" style="stop-color:#D9B879"/>
   <stop offset="0.301961" style="stop-color:#765713"/>
   <stop offset="0.509804" style="stop-color:#FFE69F"/>
   <stop offset="0.721569" style="stop-color:#906420"/>
   <stop offset="0.870588" style="stop-color:#D9BB7B"/>
   <stop offset="1" style="stop-color:#906210"/>
</linearGradient>
Additionally, I want to use these specific colors from the gradient:
"golden-gradient": [
   "#C29439",
   "#D9B879",
   "#765713",
   "#FFE69F",
   "#906420",
   "#D9BB7B",
   "#906210"
],
My goal is to apply these colors not only to the logo but also to the text on the front page of my website. Please let me know if there are any further details I might have missed in my explanation.

**Reply by wongjn** [reply]
So it seems like you are trying to define a custom gradient for the text. You'd do this with a [custom `backgroundImage` value](https://tailwindcss.com/docs/background-image#customizing-your-theme): https://play.tailwindcss.com/wf2XLmc7yT

**Reply by kaminski-dev** [reply]
This is what I need but when I adapt it into my repo only the borders of the letters have this gradient.

**Reply by kaminski-dev** [reply]
<img width="244" alt="Bildschirmfoto 2023-08-23 um 18 36 26" src="https://github.com/tailwindlabs/tailwindcss/assets/39885406/d816d540-caf8-4149-9c76-75bf27d16db3">

**Reply by wongjn** [reply]
Consider providing a [Tailwind Play](https://play.tailwindcss.com) that reproduces the unexpected behavior.

**Reply by kaminski-dev** [reply]
https://play.tailwindcss.com/MTfOndBd9R 

Here is the tailwind play link

**Reply by kaminski-dev** [reply]
<img width="474" alt="Bildschirmfoto 2023-08-23 um 18 48 53" src="https://github.com/tailwindlabs/tailwindcss/assets/39885406/76889e7c-bcb4-4d93-bc01-2195b7691f08">

**Reply by wongjn** [reply]
Your Tailwind configuration removes the `transparent` color so `text-transparent` does not apply `color: transparent`. You can work-around this either by adding `transparent: 'transparent'` to your `colors` or use an arbitrary value class `text-[transparent]`: https://play.tailwindcss.com/OVXeMWnzse

**Reply by kaminski-dev** [reply]
Thank you so much, I guess I would never come to this conclusion. 
Really appriciate your help!
